README.md

```diff
@@ -74,106 +98,37 @@ And here is the equivalent implementation in Jsonnet with
 }
 ```
 
-Using the Jsonnet command line, we can easily generate a
-`deployment.json` file from the above, which can then be sent to the
-cluster directly by `kubectl`:
+Save the file as `helloworld.libsonnet`, then run:
 
 ```bash
-$ jsonnet hello.jsonnet -m .         # Generates `deployment.json`.
-$ kubectl create -f deployment.json  # Dispatch to run on cluster.
+jsonnet helloword.libsonnet
 ```
 
-This is nice, but sometimes the default `Deployment` object is not
-precisely what you want.
+This command creates the `deployment.json` file that the 
+**ksonnet** snippet defines.
 
-A core goal of `kube.libsonnet` is to maintain the flexibility and
-expressiveness of the original Kubernetes API objects. To give you
-some idea of how easy it is to modify these objects, let's
-change the deployment specification to use a rolling update strategy 
-and a custom selector (see also [source][v2hellojsonnet]). We use
-`kube.libsonnet`'s mixins to make the changes.
+You can now apply this deployment to your Kubernetes cluster
+by running the following command:
 
-```c++
-// hello.jsonnet; imports omitted
-{
-  local nginxContainer =
-    container.Default("nginx", "nginx:1.7.9") +
-    container.NamedPort("http", 80),
-
-  "deployment.json":
-    deployment.FromContainer("nginx-deployment", 2, nginxContainer) +
-    deployment.mixin.spec.RollingUpdateStrategy() +
-    deployment.mixin.spec.Selector({ "app": "nginx" }),
-}
+```bash
+kubectl apply -f deployment.json
 ```
 
-Here we customize the `strategy` and `selector` fields, but we
-can use this method to customize any field in the Kubernetes-standard
-[Deployment Spec API object][deploymentspec]. In fact, as
-the [tutorial][tutorial] demonstrates, _all_ `kube.libsonnet` 
-objects are customizable.
+## The **ksonnet** libraries
 
-If you look at the generated YAML, you can see the customized API
-objects.
+The **ksonnet** libraries provide sets of different methods for 
+creating and manipulating Kubernetes objects:
 
-```yaml
-apiVersion: extensions/v1beta1
-kind: Deployment
-metadata:
-  name: nginx-deployment
-spec:
-  replicas: 2
-  strategy:
-    rollingUpdate:
-        maxSurge: 1,
-        maxUnavailable: 1
-    type: RollingUpdate
-  selector:
-    matchLabels:
-        app: nginx
-  template:
-    metadata:
-      labels:
-        app: nginx
-    spec:
-      containers:
-      - name: nginx
-        image: nginx:1.7.9
-        ports:
-        - containerPort: 80
-```
+* `kube/core.libsonnet`: extends the object model and functions of `Jsonnet` to implement the Kubernetes API
+* `kube/util.libsonnet`: contains methods to help create complex Kubernetes objects out of smaller objects
+
+Kubernetes v1 and v1beta1 are supported.
+
+For more examples and a fuller explanation, see the [tutorial][tutorial].
+
+## Contributing
```

Comment:
Section of the README at first, if it ends up growing we can split it off in the future.